File: Data Structures & Algorithms/climbing-stairs/submission-4.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        last_2, last_1 = 1, 2
        for _ in range(n - 2):
            tmp = last_2 + last_1
            last_2 = last_1
            last_1 = tmp
        return last_1

File: Data Structures & Algorithms/house-robber/submission-4.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int size = nums.size();
        if (size == 1) return nums[0];
        if (size == 2) return max(nums[0], nums[1]);
        int last2 = nums[0], last1 = max(nums[0], nums[1]);
        int curr_max = 0;
        for (int i = 2; i < size; ++i) {
            curr_max = max(last2 + nums[i], last1);
            last2 = last1;
            last1 = curr_max;
        }
        return curr_max;
    }
};
